File: backend/app/api/v1/assets.py
from typing import List
from uuid import UUID
from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession
from app.api.deps import get_current_user, get_db
from app.models.user import User
from app.schemas.asset import AssetResponse, BrandColorsRequest, BrandFontRequest, BrandStyleRequest
from app.schemas.common import SuccessResponse
from app.services.asset_service import AssetService

router = APIRouter()

@router.get("/{client_id}", response_model=SuccessResponse[List[AssetResponse]])
async def get_assets(client_id: UUID, db: AsyncSession = Depends(get_db), current_user: User = Depends(get_current_user)):
    return SuccessResponse(data=await AssetService(db).get_client_assets(client_id, current_user.id))

@router.post("/{client_id}/logo", response_model=SuccessResponse[AssetResponse])
async def upload_logo(client_id: UUID, db: AsyncSession = Depends(get_db), current_user: User = Depends(get_current_user)):
    return SuccessResponse(data=await AssetService(db).add_file_asset(client_id, "logo", f"/uploads/{client_id}/logo.png", current_user.id))

@router.post("/{client_id}/guideline", response_model=SuccessResponse[AssetResponse])
async def upload_guideline(client_id: UUID, db: AsyncSession = Depends(get_db), current_user: User = Depends(get_current_user)):
    return SuccessResponse(data=await AssetService(db).add_file_asset(client_id, "guideline", f"/uploads/{client_id}/guideline.pdf", current_user.id))

@router.post("/{client_id}/references", response_model=SuccessResponse[AssetResponse])
async def upload_reference(client_id: UUID, db: AsyncSession = Depends(get_db), current_user: User = Depends(get_current_user)):
    return SuccessResponse(data=await AssetService(db).add_file_asset(client_id, "reference", f"/uploads/{client_id}/reference.png", current_user.id))

@router.post("/{client_id}/fonts", response_model=SuccessResponse[AssetResponse])
async def add_font(client_id: UUID, data: BrandFontRequest, db: AsyncSession = Depends(get_db), current_user: User = Depends(get_current_user)):
    return SuccessResponse(data=await AssetService(db).add_font(client_id, data, current_user.id))

@router.delete("/{client_id}/fonts/{font_id}")
async def remove_font(client_id: UUID, font_id: UUID, db: AsyncSession = Depends(get_db), current_user: User = Depends(get_current_user)):
    await AssetService(db).remove_font(client_id, font_id, current_user.id)
    return SuccessResponse(data={"message": "Font removed"})

@router.put("/{client_id}/colors", response_model=SuccessResponse[AssetResponse])
async def update_colors(client_id: UUID, data: BrandColorsRequest, db: AsyncSession = Depends(get_db), current_user: User = Depends(get_current_user)):
    return SuccessResponse(data=await AssetService(db).update_brand_colors(client_id, data, current_user.id))

@router.put("/{client_id}/style", response_model=SuccessResponse[AssetResponse])
async def update_style(client_id: UUID, data: BrandStyleRequest, db: AsyncSession = Depends(get_db), current_user: User = Depends(get_current_user)):
    return SuccessResponse(data=await AssetService(db).update_brand_style(client_id, data, current_user.id))

File: frontend/src/api/assets.ts
import api from "@/lib/api"
import { ApiResponse } from "@/types/api"
import { BrandAsset } from "@/types/client"

export const assetsApi = {
  list: async (clientId: string): Promise<ApiResponse<BrandAsset[]>> => {
    const res = await api.get(`/assets/${clientId}`)
    return res.data
  },
  uploadLogo: async (clientId: string, file: File): Promise<ApiResponse<BrandAsset>> => {
    const formData = new FormData()
    formData.append("file", file)
    const res = await api.post(`/assets/${clientId}/logo`, formData, { headers: { "Content-Type": "multipart/form-data" } })
    return res.data
  },
  uploadGuideline: async (clientId: string, file: File): Promise<ApiResponse<BrandAsset>> => {
    const formData = new FormData()
    formData.append("file", file)
    const res = await api.post(`/assets/${clientId}/guideline`, formData, { headers: { "Content-Type": "multipart/form-data" } })
    return res.data
  },
  uploadReference: async (clientId: string, file: File): Promise<ApiResponse<BrandAsset>> => {
    const formData = new FormData()
    formData.append("file", file)
    const res = await api.post(`/assets/${clientId}/references`, formData, { headers: { "Content-Type": "multipart/form-data" } })
    return res.data
  },
  addFont: async (clientId: string, fontName: string, fontType: string): Promise<ApiResponse<BrandAsset>> => {
    const res = await api.post(`/assets/${clientId}/fonts`, { font_name: fontName, font_type: fontType })
    return res.data
  },
  removeFont: async (clientId: string, fontId: string): Promise<ApiResponse<{ message: string }>> => {
    const res = await api.delete(`/assets/${clientId}/fonts/${fontId}`)
    return res.data
  },
  updateColors: async (clientId: string, colors: string[]): Promise<ApiResponse<BrandAsset>> => {
    const res = await api.put(`/assets/${clientId}/colors`, { colors })
    return res.data
  },
  updateStyle: async (clientId: string, style: string): Promise<ApiResponse<BrandAsset>> => {
    const res = await api.put(`/assets/${clientId}/style`, { style })
    return res.data
  },
}
